src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -98,239 +102,151 @@ public static OnnxModel CreateFromBytes(byte[] modelBytes)
+            var map = OnnxToMlNetTypeMap();
+            if (!map.ContainsKey(type))
+               throw Contracts.ExceptNotSupp("Onnx type not supported", type);
+            return PrimitiveType.FromKind(map[type]);
         }
 
-        public static PrimitiveType OnnxToMlNetType(DataType type)
+        internal static Dictionary<System.Type, DataKind> OnnxToMlNetTypeMap()
         {
-            DataKind kind;
-            switch (type)
+            if (_typeToKindMap == null)
             {
-                case DataType.Type_Float:
-                    kind = DataKind.R4;
-                    break;
-
-                case DataType.Type_Double:
-                    kind = DataKind.R8;
-                    break;
-
-                case DataType.Type_Int8:
-                    kind = DataKind.I1;
-                    break;
-
-                case DataType.Type_Int16:
-                    kind = DataKind.I2;
-                    break;
-
-                case DataType.Type_Int32:
-                    kind = DataKind.I4;
-                    break;
-
-                case DataType.Type_Int64:
-                    kind = DataKind.I8;
-                    break;
-
-                case DataType.Type_Uint8:
-                    kind = DataKind.U1;
-                    break;
-
-                case DataType.Type_Uint16:
-                    kind = DataKind.U2;
-                    break;
-
-                case DataType.Type_String:
-                    kind = DataKind.TX;
-                    break;
-
-                case DataType.Type_Bool:
-                    kind = DataKind.BL;
-                    break;
-
-                case DataType.Type_Invalid:
-                default:
-                    throw Contracts.ExceptNotSupp("Onnx type not supported", type);
+                _typeToKindMap = new Dictionary<System.Type, DataKind>
+                {
+                    { typeof(Single) , DataKind.R4},
+                    { typeof(Double) , DataKind.R8},
+                    { typeof(Int16) , DataKind.I2},
+                    { typeof(Int32) , DataKind.I4},
+                    { typeof(Int64) , DataKind.I8},
+                    { typeof(UInt16) , DataKind.U2},
+                    { typeof(UInt32) , DataKind.U4},
+                    { typeof(UInt64) , DataKind.U8},
+                    { typeof(String) , DataKind.TX},
+                    { typeof(Boolean) , DataKind.BL},
+                };
             }
-
-            return PrimitiveType.FromKind(kind);
+            return _typeToKindMap;
         }
 
-        internal static Dictionary<System.Type, DataType> SystemTypeToOnnxType()
+        internal static Dictionary<System.Type, System.Type> SystemTypeToOnnxType()
         {
-            if (_typeMap == null)
+            if (_onnxTypeMap == null)
             {
-                _typeMap = new Dictionary<System.Type, DataType>
+                _onnxTypeMap = new Dictionary<System.Type, System.Type>
```

Comment:
?? This map seems pointless. It maps the a Type to itself.

Reply:
Changed it to a hash set instead. We don't need a map here. The original idea was to have a mapping from System.Type to the OnnxRuntime type system. But Onnxruntime uses System.Type now.

---
In reply to: [236342037](https://github.com/dotnet/machinelearning/pull/1717#discussion_r236342037) [](ancestors = 236342037)